Admin/Get-MigrationReport.ps1

```diff
@@ -0,0 +1,236 @@
+            Add-Content -Path $LogFile -Value " [Error] The Migration EndPoint not exist."
+        }
+        Get-MigrationConfig | Export-Clixml "$OutputFolder\MigrationConfig.xml"
+        Add-Content -Path $LogFile -Value " [INFO] The Migration Config Report has been generated successfully."
+
+        $MailboxStatistics | Export-Clixml "$OutputFolder\MailboxStatistics_$Mailbox.xml"
+        $MailboxStatistics.MoveHistory[0] | Export-Clixml "$OutputFolder\MoveReport-History.xml"
+        Add-Content -Path $LogFile -Value " [INFO] The Move Request History Report has been generated successfully."
+    } catch {
+        Add-Content -Path $LogFile -Value '[ERROR] Unable to export the Reports.'
+        Add-Content -Path $LogFile -Value $_
+        throw
+    }
+}
+
+function Export-Summary {
+    #check the log file
+    if (-not (Test-Path -Path $LogFile -ErrorAction Stop )) {
+        # Create a new log file if not found.
+        New-Item $LogFile  -Type File -Force  -ErrorAction SilentlyContinue
+    }
+
+    try {
+        if (-not (Test-Path -Path $file -ErrorAction Stop )) {
+            # Create a new log file if not found.
+            New-Item $file   -Type File -Force -ErrorAction SilentlyContinue
+        }
+    } catch {
+        Add-Content -Path $LogFile -Value '[ERROR] Unable to Create Summary File.'
+        Add-Content -Path $LogFile -Value $_
+        throw
+    }
+    $File = "Text-Summary.txt"
+    $UniqueFailure = $MoveRequestStatistics.Report.Failures | Select-Object FailureType -Unique
+    $detailedFailure = foreach ($U in $UniqueFailure) { $MoveRequestStatistics.Report.Failures | Where-Object { $_.FailureType -like $U.FailureType } | Select-Object Timestamp, FailureType, FailureSide, Message -Last 1 | Format-List }
+    New-Item $File -Type file -Force
+    "This Move Request has the following infomration:" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "the status of this Move Request is " + $MoveRequestStatistics.Status.ToString() + " with " + $MoveRequestStatistics.Status + " Percent"  >> ($File)
+    "" >> ($File)
+    $MoveRequestStatistics.Message.ToString() >> ($File)
+    "" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "The Move Request has the following Failures:" >> ($File)
+    $MoveRequestStatistics.Report.Failures | Group-Object  FailureType | Format-Table Count, Name >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "Here is more details about each Failure (Note that only the last error is selected in more details):" >> ($File)
+    "" >> ($File)
+    $detailedFailure >> ($File)
+    Add-Content -Path $LogFile -Value "[INFO] the summary report has been created successfully."
+}
+Function Find-FailureMessage {
+    param (
+        [string]$failureMessage = $FailureMessages[0].Message
```

Comment:
Don't set the parameter by default from a script variable. Let's just pass it in. Also, do we only care about the 1st type of exception or all unique errors?

Reply:
Though it would be better to check all the unique exceptions,  it was difficult to compare arrays against strings with if statment.